### DISCUSSION TITLE
Error executing s1_testmodel.i | Registering objects

### DISCUSSION THREAD

**Discussion by rl3fz** [idaholab/moose#18041] [Q&A Modules: Phase field] [answered]
Hello, I'm attempting to execute the 's1_testmodel.i' file for spinodal decomposition (Step 1 of the phase field tutorial), but I keep getting:

> *** ERROR ***
> A 'SplitCHWRes' is not a registered object.
> 
> If you are trying to find this object in a dynamically linked library, make sure that
> the library can be found either in your "Problem/library_path" parameter or in the
> MOOSE_LIBRARY_PATH environment variable.

I did not encounter a similar issue with the simple_diffusion or pressure_diffusion input files, and I did not need to register any additional objects for these. I've looked at the documentation for ADSplitCHWRes, but am not sure if or where I would need to incorporate it. The Getting Started Tutorial section on Objects states "All MOOSE applications posses the full capability of the framework and physics modules," so I am wary of creating duplications/any unnecessary files.

Any advice would be greatly appreciated.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You probably did not link the phase field module in the executable you are using.
Could you please look for 'phase field' in the Makefile that was used for compiling the executable?

All MOOSE applications can get all physics modules, by default we only include the bare minimum.

Guillaume

**Reply by rl3fz** [reply]
Yes, ok I've changed the "no" next to PHASE_FIELD to "yes", saved the file, rebooted the terminal, and tried it again but got the same error.
I then added `# include "ADSplitCHWRes.h"` to the top of the input file but still got the same error.

**Reply by GiudGiud** [reply]
You need to recompile. 
Rebooting the terminal does not change the executable
You do not need to add include code headers in input files

**Reply by rl3fz** [reply]
I can run the file now, thank you!
